profiling/variorum-connector/variorum-connector.cpp

```diff
@@ -14,425 +14,220 @@
+      break;
+    case Kokkos::Tools::Experimental::DeviceType::OpenMP:
+      return "OPENMP";
+      break;
+    case Kokkos::Tools::Experimental::DeviceType::Cuda: return "CUDA"; break;
+    case Kokkos::Tools::Experimental::DeviceType::HIP: return "HIP"; break;
+    case Kokkos::Tools::Experimental::DeviceType::OpenMPTarget:
+      return "OPENMPTARGET";
+      break;
+    case Kokkos::Tools::Experimental::DeviceType::HPX: return "HPX"; break;
+    case Kokkos::Tools::Experimental::DeviceType::Threads:
+      return "THREADS";
+      break;
+    case Kokkos::Tools::Experimental::DeviceType::SYCL: return "SYCL"; break;
+    case Kokkos::Tools::Experimental::DeviceType::OpenACC:
+      return "OPENACC";
+      break;
+    default: return "UNKOWN";
   }
-  time_t total_time;
-  time_t end_time;
-  time(&end_time);
-  std::cout << "End Time: " << end_time << "\nStart Time: " << start_time
-            << "\n";
-  total_time = end_time - start_time;
-
-  std::cout << "The kokkos library was alive for " << total_time << " seconds."
-            << std::endl;
 }
 
 void kokkosp_begin_parallel_for(const char* name, const uint32_t devID,
                                 uint64_t* kID) {
-  std::cout << "Device ID: " << devID << "\n";
-  if (usingMPI) {
-    variorum_call_mpi();
-  } else {
-    variorum_call();
-  }
+  auto result = Kokkos::Tools::Experimental::identifier_from_devid(devID);
+  global_kernel_name = std::string(name);
+  global_instance_id = result.instance_id;
+  global_device_type = device_type_to_string(result.type);
+  global_device_id   = result.device_id;
+  variorum_call();
 }
 
-void kokkosp_end_parallel_for(const uint64_t kID) {
-  if (usingMPI) {
-    variorum_call_mpi();
-  } else {
-    variorum_call();
-  }
-}
+void kokkosp_end_parallel_for(const uint64_t kID) { variorum_call(); }
 
 void kokkosp_begin_parallel_scan(const char* name, const uint32_t devID,
                                  uint64_t* kID) {
-  std::cout << "Device ID: " << devID << "\n";
-  if (usingMPI) {
-    variorum_call_mpi();
-  } else {
-    variorum_call();
-  }
+  auto result = Kokkos::Tools::Experimental::identifier_from_devid(devID);
+  global_kernel_name = std::string(name);
+  global_instance_id = result.instance_id;
+  global_device_type = device_type_to_string(result.type);
+  global_device_id   = result.device_id;
+  variorum_call();
 }
 
-void kokkosp_end_parallel_scan(const uint64_t kID) {
-  if (usingMPI) {
-    variorum_call_mpi();
-  } else {
-    variorum_call();
-  }
-}
+void kokkosp_end_parallel_scan(const uint64_t kID) { variorum_call(); }
 
 void kokkosp_begin_parallel_reduce(const char* name, const uint32_t devID,
                                    uint64_t* kID) {
-  std::cout << "Device ID: " << devID << "\n";
-  if (usingMPI) {
-    variorum_call_mpi();
-  } else {
-    variorum_call();
-  }
+  auto result = Kokkos::Tools::Experimental::identifier_from_devid(devID);
+  global_kernel_name = std::string(name);
```

Comment:
```suggestion
  global_kernel_name = name;
```
Same comment everywhere